skrub/_interpolation_joiner.py

```diff
@@ -0,0 +1,343 @@
+        them to the scikit-learn estimators. These transformations are
+        necessary when we are joining on columns that cannot be used directly,
+        such as timestamps or strings representing high-cardinality categories.
+
+    estimators_ : list of dicts
+        The estimators used to infer values to be joined. Each entry in this
+        list is a dictionary with keys ``"estimator"`` (the fitted estimator)
+        and ``"columns"`` (the list of columns in ``aux_table`` that it is
+        trained to predict).
+
+    See Also
+    --------
+    Joiner :
+        Works in a similar way but instead of inferring values, picks the
+        closest row from the auxiliary table.
+
+    Examples
+    --------
+    >>> buildings
+       latitude  longitude  n_stories
+    0       1.0        1.0          3
+    1       2.0        2.0          7
+
+    >>> annual_avg_temp
+       latitude  longitude  avg_temp
+    0       1.2        0.8      10.0
+    1       0.9        1.1      11.0
+    2       1.9        1.8      15.0
+    3       1.7        1.8      16.0
+    4       5.0        5.0      20.0
+
+    >>> InterpolationJoiner(
+    ...     annual_avg_temp,
+    ...     key=["latitude", "longitude"],
+    ...     regressor=KNeighborsRegressor(2),
+    ... ).fit_transform(buildings)
+       latitude  longitude  n_stories  avg_temp
+    0       1.0        1.0          3      10.5
+    1       2.0        2.0          7      15.5
+    """
+
+    def __init__(
+        self,
+        aux_table,
+        *,
+        main_key=None,
+        aux_key=None,
+        key=None,
+        suffix="",
+        regressor=None,
+        classifier=None,
+        n_jobs=1,
+    ):
+        self.aux_table = aux_table
+        self.main_key = main_key
+        self.aux_key = aux_key
+        self.key = key
+        self.suffix = suffix
+        self.regressor = regressor
+        self.classifier = classifier
+        self.n_jobs = n_jobs
+
+    def fit(self, X=None, y=None):
+        """Fit estimators to the `aux_table` provided during initialization.
+
+        `X` and `y` are for scikit-learn compatibility and they are
+        ignored.
+
+        Parameters
+        ----------
+        X : array-like
+            Ignored.
+
+        y : array-like
+            Ignored.
+
+        Returns
+        -------
+        self : InterpolationJoiner
+            Returns self.
+        """
+        del X, y
+        self._check_inputs()
+        key_values = self.vectorizer_.fit_transform(self.aux_table[self._aux_key])
+        estimators = self._get_estimator_assignments()
+        self.estimators_ = joblib.Parallel(self.n_jobs)(
+            joblib.delayed(_fit)(
+                key_values,
+                self.aux_table[assignment["columns"]],
+                assignment["estimator"],
+            )
+            for assignment in estimators
+        )
+        return self
+
+    def _check_inputs(self):
+        self.vectorizer_ = TableVectorizer()
```

Comment:
I can't remember our discussion about the vectorizer as an input. Why remove it? This would allow building the `Joiner` from the `InterpolationJoiner` rather than duplicating the logic.

Reply:
Now that I think about it I'm not sure why I removed it actually 😂. I added it back now